genCode.py

```diff
@@ -77,10 +77,24 @@ def createStaticClient(name, api, genAsync=False):
 
     lines.append('    classOptions = {')
 
-    copiedOptions = ('baseUrl', 'exchangePrefix')
+    # if serviceName is not set, get it from the baseUrl, exchangePrefix, or manifest name
+    # this can be removed when all services have serviceName
+    if 'serviceName' not in api:
+        if 'baseUrl' in api:
+            api['serviceName'] = api['baseUrl'].split('/')[2].split('.')[0]
+        elif 'exchangePrefix' in api:
+            api['serviceName'] = api['exchangePrefix'].split('/')[1].replace('taskcluster-', '')
```

Comment:
If the pattern is `exchange/<serviceName>/<version>`, should that be added into taskcluster-lib-urls?

Reply:
It's not needed, because the full path is embedded in the exchanges.json -- rootUrl doesn't affect it.